src/routes/v2/projects.rs

```diff
@@ -1121,6 +1122,32 @@ pub async fn project_edit(
                 .await?;
             }
 
+            // check new description and body for links to associated images
+            // if they no longer exist in the description or body, delete them
+            let uploaded_images = database::models::Image::get_many_contexted(
+                models::images::ImageContext::Project {
+                    project_id: Some(id.into()),
+                },
+                &mut transaction,
+            )
+            .await?;
+            for image in uploaded_images {
+                let description_contains = if let Some(description) = &new_project.description {
+                    description.contains(&image.url)
+                } else {
+                    false
+                };
+                let body_contains = if let Some(body) = &new_project.body {
+                    body.contains(&image.url)
+                } else {
+                    false
+                };
+                if !description_contains && !body_contains {
+                    image_item::Image::remove(image.id, &mut transaction, &redis).await?;
+                    image_item::Image::clear_cache(image.id, image.url, &redis).await?;
+                }
+            }
+
```

Comment:
I think this would be good to make this a function as threeish places call something similar